### PR TITLE
SDIT-921 PUT /api/offender-sentences/booking/{bookingId}/home-detenti…

### PR DESCRIPTION
**Pull request by steverendell**
…on-curfews/latest/*

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/repository/OffenderCurfewRepository.java [hunk 1]
```diff
@@ -69,6 +80,7 @@ public Collection<OffenderCurfew> offenderCurfews(final Set<String> agencyIds) {
 
 
     public void setHDCChecksPassed(final long curfewId, final HdcChecks hdcChecks) {
+        getCurfewLock(curfewId);
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/OffenderCurfewRepository.java:83]: Was it not possible to combine the "FOR UPDATE WAIT" in the actual SQL about to be executed? E.g change the SQL in say OffenderCurfewRepositorySql.UPDATE_CURFEW_CHECKS_PASSED ?

File: src/main/java/uk/gov/justice/hmpps/prison/repository/OffenderCurfewRepository.java [hunk 2]
```diff
@@ -186,20 +202,73 @@ public Optional<HomeDetentionCurfew> getLatestHomeDetentionCurfew(long bookingId
...
+        } catch (DataAccessException e) {
+            log.error("Error getting lock", e);
+            throw new DatabaseRowLockedException("Failed to get HDC_STATUS_REASONS lock for curfew id " + curfewId);
+        }
+    }
+
+    @NotNull
+    private String getWaitClauseIfNotHsqldb() {
+        final HikariDataSource resolvedDefaultDataSource = (HikariDataSource)routingDataSource.getResolvedDefaultDataSource();
+        return resolvedDefaultDataSource.getJdbcUrl().contains("hsqldb") ? "" : " WAIT " + lockWaitTime;
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/OffenderCurfewRepository.java:272]: very minor - I think I would have done this nasty if hsqldb stuff using Spring beans, like the way we swap certain beans depending if we are using NOMIS or hsqldb e.g https://github.com/ministryofjustice/prison-api/blob/7aceba7c8c6b9ebd675f0fb259c4c56f0f2acd63/src/main/java/uk/gov/justice/hmpps/prison/service/receiveandtransfer/TrustAccountService.kt

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/prison/repository/OffenderCurfewRepository.java (hunk 1 above)
Comment (steverendell) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/OffenderCurfewRepository.java:83]: Do you mean combine the select for update with the update statement in the sql string?

File: src/main/java/uk/gov/justice/hmpps/prison/repository/OffenderCurfewRepository.java (hunk 2 above)
Comment (steverendell) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/OffenderCurfewRepository.java:272]: Ah ok yes Ill do it the same way as this

**Review by andymarke — APPROVED** [review]
lgtm

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**